scanner/cmd/scannerctl/scale.go

```diff
@@ -208,3 +368,55 @@ func doWithTimeout(ctx context.Context, timeout time.Duration, f func(context.Co
 	defer cancel()
 	return f(ctx)
 }
+
+// profileForever queries the scanner at the given endpoint with the given client
+// and saves the contents in the given directory.
+// The stopC channel signals the profiler should terminate gracefully.
+//
+// This function writes to the stopC channel to indicate when it has terminated gracefully.
+func profileForever(service, svcAddr string, cli *http.Client, dir string, stopC chan any) {
```

Comment:
Looking back at this, I think perhaps a cancel context would probably be more idiomatic, but since this is just testing, I don't think I care enough to think about it anymore than that